app/models/night_finder.rb

```diff
@@ -0,0 +1,30 @@
+class NightFinder
+  def find_nights(night_params)
+    @_night_params = night_params
+
+    Night.all_for_couch(couch_id)
+      .between_check_in_check_out(check_in, check_out)
```

Comment:
could be as simple as the .available? behavior using enums give us? Not sure if thats best use for checking if a night is reserved or not though.